src/lsptoolshost/dotnetRuntimeExtensionResolver.ts

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { HostExecutableInformation } from "../shared/constants/HostExecutableInformation";
+import { IHostExecutableResolver } from "../shared/constants/IHostExecutableResolver";
+import { getServerPath } from "../lsptoolshost/roslynLanguageServer";
+import { PlatformInformation } from "../shared/platform";
+import { Options } from "../shared/options";
+import { existsSync } from 'fs';
+import { CSharpExtensionId } from '../constants/CSharpExtensionId';
+
+export const DotNetRuntimeVersion = '7.0';
+
+interface IDotnetAcquireResult {
+    dotnetPath: string;
+}
+
+/**
+ * Resolves the dotnet runtime from given options and the dotnet runtime VSCode extension.
+ */
+export class DotnetRuntimeExtensionResolver implements IHostExecutableResolver {
```

Comment:
moved the runtime resolution into the `IHostExecutableResolver` so we can get this information in other places (and share more code with O#).

Reply:
this is all essentially just moved